packages/um/package.py

```diff
@@ -0,0 +1,138 @@
+    version("13.6", revision=124981)
+
+    maintainers("penguian")
+
+    variant("optim", default="safe", description="Optimization level",
+        values=("debug", "high", "rigorous", "safe"), multi=False)
+    variant("platform", default="nci-x86-ifort", description="Site platform",
+        values=("nci-x86-ifort", "vm-x86-gnu"), multi=False)
+
+
+    depends_on("fcm", type="build")
+    depends_on("openmpi@4.0.2:4.1.0", type=("build", "run"))
+    depends_on("gcom", type=("build", "link"))
+    depends_on("eccodes +fortran +netcdf", type=("build", "link"))
+    depends_on("netcdf-fortran@4.5.2", type=("build", "link"))
+
+    phases = ["build", "install"]
+
+
+    def _get_linker_args(self, spec, libname):
+        """
+        The reason for the explicit -rpath is:
+        https://github.com/ACCESS-NRI/spack_packages/issues/14#issuecomment-1653651447
+        """
+        ld_flags = spec[libname].libs.ld_flags
+        rpaths = ["-Wl,-rpath=" + d for d in spec[libname].libs.directories]
+        return " ".join([ld_flags] + rpaths)
+
+
+    def setup_build_environment(self, env):
+        """
+        Set environment variables to their required values.
+        """
+        spec = self.spec
+        env.prepend_path("PATH", spec["fcm"].prefix.bin)
+        ideps = ["eccodes", "gcom", "netcdf-fortran"]
+        incs = [spec[d].prefix.include for d in ideps]
+        for ipath in incs:
+            env.prepend_path("CPATH", ipath)
+        """
+        The gcom library does not contain shared objects and
+        therefore must be statically linked.
+        """
+        env.prepend_path("LIBRARY_PATH", spec["gcom"].prefix.lib)
+
+        # Set configuration options
+        config = configparser.ConfigParser()
+        config.read(join_path(self.package_dir, "rose-app.conf"))
+        for key in config["env"]:
+            if len(key) > 0 and key[0] != '!':
+                value = config["env"][key].replace("\n=", "\n")
+                env.set(key, value)
+
+        # Override some specific environment variables
+        env.set("optimisation_level", spec.variants["optim"].value)
+        env.set("platform_config_dir", spec.variants["platform"].value)
+        env.set("um_rev", f"vn{spec.version}")
+        components = ["casim", "jules", "shumlib", "socrates", "ukca"]
+        for comp in components:
+            key = f"{comp}_rev"
+            value = f"um{spec.version}"
+            env.set(key, value)
+        dep = {
+                "drhook": "drhook",
+                "eccodes": "eccodes",
+                "netcdf": "netcdf-fortran"}
+        extra_ld_flags = {
+                "drhook": "-ldrhook",
+                "eccodes": "-leccodes_f90 -leccodes",
+                "netcdf": "-lnetcdff -lnetcdf"}
+        for libname in ["eccodes", "netcdf"]:
+            linker_args = " ".join([
+                    extra_ld_flags[libname],
+                    self._get_linker_args(spec, dep[libname])])
```

Comment:
Could all this be completed in `_get_linker_args()`?

Reply:
Merged into `_get_linker_args()`